core/job/service/job_service.go

```diff
@@ -552,30 +619,43 @@ func (j *JobService) RefreshResourceDownstream(ctx context.Context, resourceURNs
 }
 
 func (j *JobService) Validate(ctx context.Context, jobTenant tenant.Tenant, jobSpecs []*job.Spec, jobNamesWithInvalidSpec []job.Name, logWriter writer.LogWriter) error {
-	me := errors.NewMultiError("validate specs errors")
-
 	tenantWithDetails, err := j.tenantDetailsGetter.GetDetails(ctx, jobTenant)
 	if err != nil {
 		j.logger.Error("error getting tenant details: %s", err)
 		return err
 	}
 
-	err = job.Specs(jobSpecs).Validate()
-	me.Append(err)
-	validatedJobSpecs := job.Specs(jobSpecs).GetValid()
+	if err := job.Specs(jobSpecs).Validate(); err != nil {
+		return err
+	}
 
+	validatedJobSpecs := job.Specs(jobSpecs).GetValid()
 	existingJobs, err := j.jobRepo.GetAllByTenant(ctx, jobTenant)
-	me.Append(err)
-
-	toAdd, toUpdate, toDelete, unmodifiedSpecs, err := j.differentiateSpecs(existingJobs, validatedJobSpecs, jobNamesWithInvalidSpec)
-	logWriter.Write(writer.LogLevelInfo, fmt.Sprintf("[%s] found %d new, %d modified, and %d deleted job specs", jobTenant.NamespaceName().String(), len(toAdd), len(toUpdate), len(toDelete)))
-	me.Append(err)
+	if err != nil {
+		return err
+	}
 
-	if len(toAdd)+len(toUpdate)+len(toDelete) == 0 {
-		return me.ToErr()
+	toAdd, toUpdate, toDelete, unmodifiedSpecs, unmodifiedDirtySpecs := j.differentiateSpecs(existingJobs, validatedJobSpecs, jobNamesWithInvalidSpec)
+	logWriter.Write(writer.LogLevelInfo, fmt.Sprintf("[%s] found %d new, %d modified, %d deleted and %d dirty job specs", jobTenant.NamespaceName().String(), len(toAdd), len(toUpdate), len(toDelete), len(unmodifiedDirtySpecs)))
```

Comment:
I don't think we need to mention the number of dirty jobs here, as it will be mentioned below. otherwise it might confuse what is `dirty` and what is `unprocessed`.